src/Thesis/Composition/PrivacyFilter.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Thesis.Composition.PrivacyFilter where
+
+import Thesis.Types
+import Thesis.ValueGuard
+
+data BudgetDepleted = BudgetDepleted
+  deriving (Eq, Show)
+
+type QueryPrice = (Positive Epsilon, NonNegative Delta)
+
+class PrivacyFilter a where
+  type Budget a :: *
+  emptyState :: Budget a -> a
+  subtractBudget :: a -> QueryPrice -> Either BudgetDepleted a
```

Comment:
Should we change the type of subtractBudget to `a -> QueryPrice -> Either BudgetDepleted (PrivacyFilter a)`?